scripts/pylib/twister/twisterlib/testinstance.py

```diff
@@ -51,7 +59,7 @@ def __init__(self, testsuite, platform, outdir):
         self.domains = None
 
         self.run = False
-        self.testcases = []
+        self.testcases: list[TestCase] = []
```

Comment:
If you use `list[...]`, then probably you should add `from __future__ import annotations` at the beginning for Python 3.8. But on the other hand CI passed, so maybe it is not so necessary...

Reply:
I will add this import. For python3.8 it is probable not required (for that case, AFAIK we used that for a conditional annotations, e.g. `self.testcases: list[TestCase] | None = None`)